### PR TITLE
Feature/offers_discounts_APIs

### PR DESCRIPTION
**Pull request by mahadoali**
This pull request introduces the implementation of the Offers and Discount APIs, which enable users to create, manage, and apply promotional offers and discounts within our application. These APIs are designed to enhance user engagement and drive sales by providing flexible discount options.
Also all new features have been tested with unit tests covering various scenarios, including edge cases for offer creation and redemption.

### CODE REVIEW

**Review by AhendaEsther — APPROVED** [review]

**Review by magretjumbedavid — APPROVED** [review]

**Review by mercywangara — APPROVED** [review]

**Review by mercywangara — APPROVED** [review]

**Review by meazamisgina — APPROVED** [review]

File: mitumbaesales/mitumbaesales/settings.py [hunk 1]
```diff
@@ -38,13 +38,18 @@
     'django.contrib.messages',
     'django.contrib.staticfiles',
     'notification',
-    'cart',    
-    'order',
-    'authentication',
-    'product',
+    # 'cart',
```
Comment (Ndichu-shee) [inline mitumbaesales/mitumbaesales/settings.py:41]: why are you commenting this?

File: mitumbaesales/notification/admin.py [hunk 2]
```diff
@@ -1,7 +1,7 @@
-from django.contrib import admin
-from django.contrib import admin
-from .models import Notification
-from authentication.models import User
-# Register your models here.
-admin.site.register(Notification)
-# admin.site.register(User)
+# from django.contrib import admin
```
Comment (Ndichu-shee) [inline mitumbaesales/notification/admin.py:1]: why do you have commented code

File: mitumbaesales/offer/models.py [hunk 3]
```diff
@@ -62,15 +62,15 @@ class Meta:
...
-
-    def __str__(self):
-        return f"Offer '{self.offer.name}' includes Product '{self.product.name}'"
-
-    class Meta:
-        verbose_name_plural = "Offer Products"
-        unique_together = (('offer', 'product'),) 
-        ordering = ['offer']
\ No newline at end of file
+# class OfferProduct(models.Model):
```
Comment (Ndichu-shee) [inline mitumbaesales/offer/models.py:65]: same here

**Review by mercywangara — APPROVED** [review]

**Review by meazamisgina — APPROVED** [review]

**Review by AhendaEsther — DISMISSED** [review]

**Review by magretjumbedavid — DISMISSED** [review]

**Review by mercywangara — DISMISSED** [review]

**Review by meazamisgina — APPROVED** [review]

File: mitumbaesales/offer/admin.py [hunk 4]
```diff
@@ -1,13 +1,13 @@
-from django.contrib import admin
-from .models import Offer, Discount
-# OfferDiscount, OfferProduct
-#from .models import Product
+# from django.contrib import admin
```
Comment (Ndichu-shee) [inline mitumbaesales/offer/admin.py:1]: why do you have commented code?

**Review by meazamisgina — APPROVED** [review]

**Review by mercywangara — APPROVED** [review]

**Review by AhendaEsther — APPROVED** [review]
